client/ayon_core/hosts/max/plugins/publish/validate_instance_in_context.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""Validate if instance asset is the same as context asset."""
+from __future__ import absolute_import
+
+import pyblish.api
+from ayon_core.pipeline.publish import (
+    RepairAction,
+    ValidateContentsOrder,
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from ayon_core.hosts.max.api.action import SelectInvalidAction
+from pymxs import runtime as rt
+
+
+class ValidateInstanceInContext(pyblish.api.InstancePlugin,
+                                OptionalPyblishPluginMixin):
+    """Validator to check if instance asset match context asset.
+
+    When working in per-shot style you always publish data in context of
+    current asset (shot). This validator checks if this is so. It is optional
+    so it can be disabled when needed.
+
+    Action on this validator will select invalid instances.
+    """
+
+    order = ValidateContentsOrder
+    label = "Instance in same Context"
+    optional = True
+    hosts = ["max"]
+    actions = [SelectInvalidAction, RepairAction]
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+
+        instance_node = rt.getNodeByName(instance.data.get(
+            "instance_node", ""))
+        if not instance_node:
+            return
+        asset = rt.getUserProp(instance_node, "folderPath")
+        context_asset = self.get_context_asset(instance)
+        task = rt.getUserProp(instance_node, "task")
+        context_task = self.get_context_task(instance)
```

Comment:
Why does this need to go over the instance node? Isn't this actually collected data already?